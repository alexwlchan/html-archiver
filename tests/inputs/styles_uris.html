<html>

<head>
  <meta charset="utf-8">
  <title>This is a page with styles that call out to uri resources</title>

  <!-- Just a straight passthrough of an image -->
  <style>
    div { background: url('rainbow.png'); }
  </style>

  <!-- A missing resource, so we should leave this untouched -->
  <style>
    div#missing { background: url('missing.jpg'); }
  </style>

  <!-- And a working resource with some attributes on the style tag -->
  <style foo="bar">
    div#with_attributes { background: url('snake.gif'); }
  </style>

  <!-- Look up the same image a second time -->
  <style>
    div#repeated { background: url('rainbow.png'); }
  </style>

  <!-- A URI value that's already a data: value -->
  <style>
    div#data { background: url(data:image/foo;base64,0123456789); }
  </style>

</head>

<body>
  <p>This is a page to test how html-archiver handles styles that use external resources.</p>

  <p>Anything with a url() that can be accessed should be inlined and base64-encoded.</p>
</body>

</html>
