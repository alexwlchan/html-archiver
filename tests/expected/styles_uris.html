<html>

<head>
  <meta charset="utf-8">
  <title>This is a page with styles that call out to uri resources</title>

  <!-- Just a straight passthrough of an image -->
  <style>
div { background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEYAAABGCAIAAAD+THXTAAAAAXNSR0IArs4c6QAAAAlwSFlzAAALEwAACxMBAJqcGAAAAjBpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIgogICAgICAgICAgICB4bWxuczp0aWZmPSJodHRwOi8vbnMuYWRvYmUuY29tL3RpZmYvMS4wLyI+CiAgICAgICAgIDx4bXA6Q3JlYXRvclRvb2w+QWNvcm4gdmVyc2lvbiA0LjUuODwveG1wOkNyZWF0b3JUb29sPgogICAgICAgICA8dGlmZjpDb21wcmVzc2lvbj41PC90aWZmOkNvbXByZXNzaW9uPgogICAgICAgICA8dGlmZjpZUmVzb2x1dGlvbj43MjwvdGlmZjpZUmVzb2x1dGlvbj4KICAgICAgICAgPHRpZmY6WFJlc29sdXRpb24+NzI8L3RpZmY6WFJlc29sdXRpb24+CiAgICAgIDwvcmRmOkRlc2NyaXB0aW9uPgogICA8L3JkZjpSREY+CjwveDp4bXBtZXRhPgpDyBsAAAAA5ElEQVRoBe2YQQ7DIAwETZWHpy9v+4WVQJOuJmfMxjNGQqzPtH2vtoZmbOkfnGpJSwgBBw/BHoYWWrrmDiE8fnmhJVt6/NT9flBLWkIIOHgI9jC00NL61D0+FFqypfCkIsu1hGAPQ7UUAkOWawnBHoau8fYQIgOWe5YA6HGklmJkQIGWAOhxpJZiZEDBNe+264ODB8xRHKmlGBlQoCUAehyppRgZUFBoaU3d40OhJVsCTnscqaUYGVCgJQB6HKmlGBlQcN1A6NlIB+8s3z27a2kPx7O7aOks3z27a2kPx7O7FFr6AhBlCArDDUhZAAAAAElFTkSuQmCC); }
</style>

  <!-- A missing resource, so we should leave this untouched -->
  <style>
    div#missing { background: url('missing.jpg'); }
  </style>

  <!-- And a working resource with some attributes on the style tag -->
  <style foo="bar">
div#with_attributes { background: url(data:image/gif;base64,R0lGODdhRgBGAJEAAAAAAP8AAACAQP///yH5BAQAAAAALAAAAABGAEYAAALKnI+py+0Po5y02ouz3rz7D4biSJbmiaaLwLbuC8fy/GL0jeeurff+fvkJZYAioMUbKlnGIyu5HDaRigAkupw+E9Yr9ssKiMcSMLgaMX8XXQdNpXnDobA5vWa3yPOUPb8889cXKAgoU2gYg5imFnXX6PMIqSM5iVNp6TeYGRnE2an32YMpihdaern4oKmqwNqK8AprIDtbC3vbmqu6u9iL+OuKmno6TFhsrOiZfLjMXOf8DIQsvUVdTWqcPTzb7f0NHi4+Tl5ufo5RAAA7); }
</style>

  <!-- Look up the same image a second time -->
  <style>
div#repeated { background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEYAAABGCAIAAAD+THXTAAAAAXNSR0IArs4c6QAAAAlwSFlzAAALEwAACxMBAJqcGAAAAjBpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIgogICAgICAgICAgICB4bWxuczp0aWZmPSJodHRwOi8vbnMuYWRvYmUuY29tL3RpZmYvMS4wLyI+CiAgICAgICAgIDx4bXA6Q3JlYXRvclRvb2w+QWNvcm4gdmVyc2lvbiA0LjUuODwveG1wOkNyZWF0b3JUb29sPgogICAgICAgICA8dGlmZjpDb21wcmVzc2lvbj41PC90aWZmOkNvbXByZXNzaW9uPgogICAgICAgICA8dGlmZjpZUmVzb2x1dGlvbj43MjwvdGlmZjpZUmVzb2x1dGlvbj4KICAgICAgICAgPHRpZmY6WFJlc29sdXRpb24+NzI8L3RpZmY6WFJlc29sdXRpb24+CiAgICAgIDwvcmRmOkRlc2NyaXB0aW9uPgogICA8L3JkZjpSREY+CjwveDp4bXBtZXRhPgpDyBsAAAAA5ElEQVRoBe2YQQ7DIAwETZWHpy9v+4WVQJOuJmfMxjNGQqzPtH2vtoZmbOkfnGpJSwgBBw/BHoYWWrrmDiE8fnmhJVt6/NT9flBLWkIIOHgI9jC00NL61D0+FFqypfCkIsu1hGAPQ7UUAkOWawnBHoau8fYQIgOWe5YA6HGklmJkQIGWAOhxpJZiZEDBNe+264ODB8xRHKmlGBlQoCUAehyppRgZUFBoaU3d40OhJVsCTnscqaUYGVCgJQB6HKmlGBlQcN1A6NlIB+8s3z27a2kPx7O7aOks3z27a2kPx7O7FFr6AhBlCArDDUhZAAAAAElFTkSuQmCC); }
</style>

  <!-- A URI value that's already a data: value -->
  <style>
    div#data { background: url(data:image/foo;base64,0123456789); }
  </style>

</head>

<body>
  <p>This is a page to test how html-archiver handles styles that use external resources.</p>

  <p>Anything with a url() that can be accessed should be inlined and base64-encoded.</p>
</body>

</html>
